.github/workflows/trivy.yml

```diff
@@ -1,46 +1,33 @@
+# Build the Docker image and scan it with Trivy
 name: dsp-appsec-trivy
-on: [pull_request]
+on:
+  workflow_dispatch: {}
+  pull_request:
 
 jobs:
   appsec-trivy:
-    # Parse Dockerfile and build, scan image if a "blessed" base image is not used
     name: DSP AppSec Trivy check
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-
-      # fetch JDK
-      - uses: actions/setup-java@v2
+      - name: Check out the code
+        uses: actions/checkout@v4
+      - name: Set up JDK 17 and cache Gradle build
+        uses: actions/setup-java@v4
         with:
           distribution: 'temurin'
           java-version: '17'
-
-      # set up Gradle cache
-      - uses: actions/cache@v3
-        with:
-          path: |
-            ~/.gradle/caches
-            ~/.gradle/wrapper
-          key:
-            gradle-${{ hashFiles('**/*.gradle*') }}
-          restore-keys: |
-            gradle-
-
-      # build the image
-      - name: Build
+          cache: 'gradle'
+      - name: Echo image name to output
         id: build
         run: |
-          # Enable pipefail so a gradle build failure fails this step.
+          # enable pipefail so a gradle build failure fails this step.
           set -o pipefail
           # build sources and store the plain log without colors
-          ./gradlew jibDockerBuild --console=plain \
-            | perl -pe 's/\x1b\[[0-9;]*[mG]//g' | tee build.log
-
+          ./gradlew jibDockerBuild --console=plain > build.log
```

Comment:
Is this correct? This change will no longer output `jib` to the build logs. if `jib` fails, wouldn't we want to see the output?

Reply:
We store output in `build.log` to extract the Docker container to get scanned. It's simpler to split this process into multiple steps. I can switch it back to `| tee build.log`